inspect/index.ts

```diff
@@ -0,0 +1,213 @@
+import type {
+  ArcjetRuleResult,
+  ArcjetRuleState,
+  ArcjetBotReason,
+  ArcjetErrorReason,
+} from "@arcjet/protocol";
+
+function isBotReason(reason: unknown): reason is ArcjetBotReason {
+  return (
+    typeof reason === "object" &&
+    reason !== null &&
+    "isBot" in reason &&
+    typeof reason.isBot === "function" &&
+    reason.isBot() &&
+    "isSpoofed" in reason &&
+    typeof reason.isSpoofed === "function" &&
+    "isVerified" in reason &&
+    typeof reason.isVerified === "function"
+  );
+}
+
+function isErrorReason(reason: unknown): reason is ArcjetErrorReason {
+  return (
+    typeof reason === "object" &&
+    reason !== null &&
+    "isError" in reason &&
+    typeof reason.isError === "function" &&
+    reason.isError() &&
+    "message" in reason &&
+    typeof reason.message === "string"
+  );
+}
+
+function isActive(
+  result: unknown,
+): result is ArcjetRuleResult & { state: Exclude<ArcjetRuleState, "DRY_RUN"> } {
+  return (
+    typeof result === "object" &&
+    result !== null &&
+    "state" in result &&
+    typeof result.state === "string" &&
+    result.state !== "DRY_RUN"
+  );
+}
+
+/**
+ * Determines if a `LIVE` bot rule detected a spoofed request. If `true`, the
+ * request was spoofed and should be denied. `DRY_RUN` rules are ignored.
+ *
+ * For `allow` rules, Arcjet verifies the authenticity of detected bots by
+ * checking IP data and performing reverse DNS lookups. This helps protect
+ * against spoofed bots where clients pretend to be someone else.
+ *
+ * Note that spoofed bot detection is not available on free plans.
+ *
+ * @param {ArcjetRuleResult} result - The rule result to inspect.
+ * @returns `true` if the rule result was `LIVE` and detected as a spoofed bot,
+ * or `false` otherwise.
+ *
+ * @example
+ * ```ts
+ * import arcjet, { detectBot } from "@arcjet/next";
+ * import { isSpoofedBot } from "@arcjet/inspect";
+ *
+ * const aj = arcjet({
+ *  key: process.env.ARCJET_KEY,
+ *  rules: [
+ *    detectBot({
+ *      mode: "LIVE",
+ *      allow: [],
```

Comment:
```suggestion
 *      allow: ["GOOGLE_CRAWLER", "BING_CRAWLER"],
```

It might be best to include some bots in the allow list, because we only run verification on bots that we are explicitly allowing.

Reply:
I made all these `"CATEGORY:SEARCH_ENGINE"` because that's what we have in most docs.